### ISSUE TITLE
errors during test run

### ISSUE THREAD

**Issue by weihongqi** [a2iEditing/RNAEditingIndexer#2]
Dear All,

I installed RNAEditingIndexer following the instruction. But got some errors while analyzing the test dataset:
/usr/local/ngseq/packages/Tools/BEDTools/2.26.0/bin/bedtools: /misc/ngseq8/src/RNAEditingIndexer/src/RNAEditingIndex/libz.so.1: version `ZLIB_1.2.3.3' not found (required by /usr/local/ngseq/p
ackages/Tools/BEDTools/2.26.0/bin/bedtools)

/usr/local/ngseq/packages/Tools/samtools/1.8/bin/samtools: symbol lookup error: /lib/x86_64-linux-gnu/libncurses.so.5: undefined symbol: _nc_putchar

both bedtools and samtools were intalled using module, compiled from source. 

The system info:
Linux 3.16.0-9-amd64 #1 SMP Debian 3.16.68-1 (2019-05-22) x86_64 GNU/Linux

Could you please give me some hints where to debug?

Many thanks in advance,
Weihong

**Comment by shalomhillelroth** [comment]
Dear Weihong,

I think the problems are with the installation of bedtools and samtools (and not with your installation of the RNAEditingIndex) and are probably runtime environment issues (two different issues though).
Regarding bedtools, in my opinion, it appears that for some reason the path of the RNAEditingIndex "exe" is preceding your normal system libraries directory when running bedtools. There should be a way (either when installing it or manipulating the $PATH) to make it uses the correct  files (it should __not__ use the files in the src directory)
Regarding samtools, it is a known issue, you can try updating libncurses of maybe installing samtools using conda.
In any case - you should be able to run both the bedtools and the samtools independently of the RNAEditingIndex. 
 
You can also try to use the docker version of RNAEditingIndex (which doesn't involve configuring the runtime environment, thus avoiding these issues).  

Best,
Hillel

**Comment by weihongqi** [comment]
Dear Hillel,

Many thanks for your quick response. You are right. I just solved the problem by creating a conda virtual env. Within the environment I installed bedtools and samtools, before installing RNAEditingIndexer. The test analysis was successful this time.

I did try to install the docker image. Unfortunately the disk partition set for our docker was very small. Although I modified the path of the "Resource" to another partition in the Dockerfile. All data files were still downloaded to /bin/AEI/ and the image installation failed due to lack to disk space. 

Thanks again.

Weihong
